onnxruntime/core/providers/coreml/builders/impl/base_op_builder.h

```diff
@@ -36,6 +36,11 @@ class BaseOpBuilder : public IOpBuilder {
   static bool IsInputDtypeSupport(const Node& node, size_t idx, const OpBuilderInputParams& input_params,
                                   const logging::Logger& logger);
 
+  // See this issue, https://github.com/apple/coremltools/issues/1003
+  // https://developer.apple.com/metal/Metal-Feature-Set-Tables.pdf has maximum texture widths which may be the
+  // root cause.
+  static bool CheckShapeForConvMemoryLimit(onnxruntime::VectorInt64& shape, const logging::Logger& logger);
```

Comment:
nit: it feels out of place in base_op_builder.h as it is not relevant to most ops. maybe move it to a separate helper function in helper.h/cc?

Reply:
Ah, fair point -- moved to the helper.h/cc files!